vault/resource_ldap_secret_backend.go

```diff
@@ -117,7 +124,7 @@ func ldapSecretBackendResource() *schema.Resource {
 	})
 
 	// Add common mount schema to the resource
-	provider.MustAddSchema(resource, getMountSchema("type"))
+	provider.MustAddSchema(resource, getMountSchema("path", "type"))
```

Comment:
Without this change `path` was being required.

Reply:
I looked at this a little more and I think use of `getMountSchema` is not right for the LDAP backend resource? For example, it adds a `max_lease_ttl_seconds` which I think maps to Vault's [max_lease_ttl](https://developer.hashicorp.com/vault/docs/configuration#max_lease_ttl). But I don't think that should be configurable on the plugin. The read on the LDAP config by the vault CLI does not return `max_lease_ttl`.

Reply:
What does `getMountSchema` do exactly? I am trying to understand why we are using it here? The only other backend resource we are using this in is for kubernetes. 🤔 
Looking at the [schema map](https://github.com/hashicorp/terraform-provider-vault/blob/744b2999b9020c482749aa95984770c5f146d6ea/vault/resource_mount.go#L20), the only required fields are `path` and `type`

Reply:
It looks like `getMountSchema` was originally written for the generic [vault_mount](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/resources/mount) resource. (I don't understand the purpose of vault_mount so I asked in Slack for some context.) But calling it here adds all those fields to this resource. I am not sure we want to do that.

cc @vinay-gopalan Wondering if you have any context on why only the kube secret backend uses this pattern and if we should use it here?

Reply:
I chatted with @benashz and I think we will remove this call to `getMountSchema` and prefer to have a `tune` block like we do with [github auth](https://github.com/hashicorp/terraform-provider-vault/blob/4346ded8e5c9cea9f3e9bed2e93416bb1e24459f/vault/resource_github_auth_backend.go#L58). But for now I will remove this

Reply:
For some additional context — there are currently two ways of setting up resources with mounts in the TFVP. Resources can manage their mounts along with the backend config as we are doing here (this would be preferred in terms of lifecycle management in the TFVP whenever possible), or they can only manage the config and have the `vault_mount` resource deal with the actual mounting of the secret engine (this is how things were done since prior to my time at HashiCorp, and it makes sense in some cases, although we are trying to move away from it whenever possible. I believe we tried managing the mount for the MongoDB Atlas engine within the resource, but we decided to go the `vault_mount` route instead because the MongoDB engine was giving us some trouble). 

With that additional context, since it seems possible to manage the mount within the LDAP resource, we decided to have this format for the LDAP Backend resource. As per your questions above @fairclothjm , the mount schema fields that are added to the resource are not read from the LDAP config read endpoint, but all those additional fields are handled by the `readMount` util used in [these lines here](https://github.com/hashicorp/terraform-provider-vault/blob/VAULT-8099/secrets/ldap/vault/resource_ldap_secret_backend.go#L230-L232). This is why all of this should work as it is set up currently.

I think it's fine to have the `path` change as done above since we don't want `path` to be Required. Maybe we can tweak `getMountSchema` to not always return a required `path`? Adding a `tune` block can also work, but I'm not sure how we would get the info for creating the initial mount, since the `tune` block will only be populated to update the mount. All of that is currently already handled by the [block of code here](https://github.com/hashicorp/terraform-provider-vault/blob/VAULT-8099/secrets/ldap/vault/resource_ldap_secret_backend.go#L132-L140). `updateMount` is currently already performing a tune.

I'd be curious to hear what you think and if the above context helps clear some of the confusion around these 2 ways 😅 I would love to hear what folks think is the best precedent to follow when we are attempting to manage the mounts within the resources!

Reply:
Thanks for the information @vinay-gopalan !

I will need to give this a little more thought. My initial reaction is that I am not in favor of overloading `getMountSchema`. But I am willing to be flexible here if you think that is the best path forward 😄 

I think if we are going to keep this pattern of using `getMountSchema`, `updateMount`, etc then we should differentiate those fields in the docs to make it clear what is configuring the plugin and what is tuning the mount.